pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -126,6 +127,39 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// RAIDVolume defines the desired configuration of volume in RAID
+type RAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	SizeGb int `json:"sizeGb,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0, 1, 1+0.
+	// +kubebuilder:validation:Enum=0,1,1+0
+	Level nodes.RAIDLevel `json:"level" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
+	Name string `json:"name,omitempty"`
+
+	// The valid values must be "hdd" or "ssd". If this is not specified, disk type will not be a criterion to find backing physical disks
+	// +kubebuilder:validation:Enum=hdd,ssd
+	DiskType *nodes.DiskType `json:"diskType,omitempty"`
+
+	// Integer, number of disks to use for the logical disk. Defaults to minimum number of disks required
+	// for the particular RAID level.
+	// +kubebuilder:validation:Minimum=1
+	NumberOfPhysicalDisks *int `json:"numberOfPhysicalDisks,omitempty"`
+}
+
+// RAIDConfig contains the configuration that are required to config RAID in Bare Metal server
+type RAIDConfig struct {
+
+	// The logical disk that will be root volume
+	RootVolume *RAIDVolume `json:"rootVolume,omitempty"`
```

Comment:
Having a separate root volume that is just another RAIDVolume seems a little awkward. I guess we can't just reference the root volume by name because the names are optional?

How does this fit in with the work happening in #400?

Reply:
Hmm, yeah. Does the order of creation for the volumes matter? Maybe we do need to include the root volume in the array and refer to the entry here somehow. Perhaps by index?